### DISCUSSION TITLE
How to implement simple math on two columns with result in third column?

### DISCUSSION THREAD

**Discussion by yujiku808** [slick/slick#2434] [Questions] [answered]
I'm a programming newbie learning Slick with Play Framework and H2 in Scala by modifying a play-slick example I downloaded.

[https://github.com/playframework/play-samples/tree/2.8.x/play-scala-slick-example](url)

In it, you enter values in a html form and it gets rendered on the webpage as a table. The basic setup works fine but I would like to add something like (col1 - col2) with the result inserted into col3.

I've been looking through the Slick and Play docs for a couple of days but I can't find an example where they do any simple math.

**Comment by nafg** [comment, marked as answer]
You can do math, Slick provides operators as extension methods on `Rep`s.

You can also define your own operators; see https://scala-slick.org/doc/3.4.0-SNAPSHOT/userdefined.html

If you want a more specific answer please make your question more specific ;)
